Client/WarFare/UIItemUpgrade.h

```diff
@@ -0,0 +1,139 @@
+﻿// CUIItemUpgrade.h: interface for the CUIItemUpgrade class.
+//
+//////////////////////////////////////////////////////////////////////
+
+#if !defined(AFX_UIItemUpgrade_H__fd9f5093_0ed3_4c08_9e31_19c40773b24d__INCLUDED_)
+#define AFX_UITRANSACTIONDLG_H__fd9f5093_0ed3_4c08_9e31_19c40773b24d__INCLUDED_
+
+#if _MSC_VER > 1000
+#pragma once
+#endif // _MSC_VER > 1000
+
+#include <N3Base/N3UIBase.h>
+#include <N3Base/N3UIArea.h>
+#include <N3Base/N3UIString.h>
+#include <N3Base/N3UIImage.h>
+#include <N3Base/N3UIButton.h>
+#include "GameProcedure.h"
+
+#include "GameDef.h"
+#include "N3UIWndBase.h"
+
+
+
+//////////////////////////////////////////////////////////////////////
+
+class CUIItemUpgrade : public CN3UIWndBase
+{
+	friend class CUIInventory;
+
+public:
```

Comment:
Why are the members public? Nothing even needs to access it externally.
They should be protected, or private.